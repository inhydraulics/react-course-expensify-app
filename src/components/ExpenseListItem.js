import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = (props) => {
	//console.log(props);
	return (
		<div>
			<Link to={`/edit/${props.id}`}>
				<p>Description: {props.description}</p>
			</Link>
			<p>Amount: {numeral(props.amount / 100).format("$0,0.00")}</p>
			<p>Created at: {moment(props.createdAt).format("MMMM Do, YYYY")}</p>
		</div>
	);
};

export default ExpenseListItem;
