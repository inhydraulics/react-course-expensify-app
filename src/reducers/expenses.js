const initialExpensesState = [];

const expensesReducer = (state = initialExpensesState, action) => {
	switch (action.type) {
		case "ADD_EXPENSE":
			return [...state, action.expense];
		case "REMOVE_EXPENSE":
			return state.filter((el) => el.id !== action.id);
		case "EDIT_EXPENSE":
			return state.map((expense) => {
				if (expense.id === action.id) {
					return {
						...expense,
						...action.dataObj,
					};
				}
				return expense;
			});
		default:
			return state;
	}
};

export default expensesReducer;
