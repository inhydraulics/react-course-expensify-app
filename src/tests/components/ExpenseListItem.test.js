import React from "react";
import { shallow } from "enzyme";
import expenses from "../fixtures/expenses";
import ExpenseListItem from "../../components/ExpenseListItem";

test("should render ExpenseListItem", () => {
	const wrapper = shallow(
		<ExpenseListItem
			description={expenses[0].description}
			amount={expenses[0].amount}
            createdAt={expenses[0].createdAt}
            id={expenses[0].id}
		/>
	);
	expect(wrapper).toMatchSnapshot();
});
