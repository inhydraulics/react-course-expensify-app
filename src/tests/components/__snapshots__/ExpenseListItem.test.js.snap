// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <p>
      Description: 
      Gum
    </p>
  </Link>
  <p>
    Amount: 
    $1.90
  </p>
  <p>
    Created at: 
    January 1st, 1970
  </p>
</div>
`;
